### PR TITLE
cmake: Remove global include of board dir

### PR DESCRIPTION
**Pull request by galak**
Remove having ${BOARD_DIR} in the global include path for Zephyr builds.
Needed to add this to the arch/posix because of how posix "boards"
define various things like interrupt handling.  Also fixed up up_squared
to build with this change.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by nashif** [comment]
Please split into multiple commits

File: boards/x86/up_squared/dts_fixup.h [hunk 1]
```diff
@@ -7,9 +7,153 @@
 /* Board level DTS fixup file */
 
 #ifdef CONFIG_SBL_FIXUP
```
Comment (nashif) [inline boards/x86/up_squared/dts_fixup.h]: can we please put this in a separate commit?

**Review by nashif — CHANGES_REQUESTED** [review]

File: boards/x86/up_squared/dts_fixup.h (hunk 1 above)
Comment (galak) [reply boards/x86/up_squared/dts_fixup.h]: Sure, separate PR?

Comment (nashif) [reply boards/x86/up_squared/dts_fixup.h]: no need for new PR

**Review by SebastianBoe — APPROVED** [review]

File: boards/x86/up_squared/dts_fixup.h (hunk 1 above)
Comment (galak) [reply boards/x86/up_squared/dts_fixup.h]: Done

**Review by aescolar — APPROVED** [review]
+1 for the board include part.

**Review by nashif — APPROVED** [review]
